docs/contributors/code/testing-overview.md

```diff
@@ -212,6 +212,86 @@ describe( 'my module', () => {
 } );
 ```
 
+### User interactions
+
+Simulating user interactions is a great way to **write tests from the user's perspective**, and therefore avoid testing implementation details.
+
+When writing tests with Testing Library, there are two main alternatives for simulating user interactions:
+
+1. The [`fireEvent`](https://testing-library.com/docs/dom-testing-library/api-events/#fireevent) API, a utility for firing DOM events part of the Testing Library core API.
+2. The [`user-event`](https://testing-library.com/docs/user-event/intro/) library, a companion library to Testing Library that simulates user interactions by dispatching the events that would happen if the interaction took place in a browser.
+
+The built-in `fireEvent` is a utility for dispatching DOM events. It dispatches exactly the events that are described in the test spec - even if those exact events never had been dispatched in a real interaction in a browser.
+
+On the other hand, the `user-event` library exposes higher-level methods (e.g. `type`, `selectOptions`, `clear`, `doubleClick`...), that dispatch events like they would happen if a user interacted with the document, and take care of any react-specific quirks.
+
+For the above reasons, **the `user-event` library is the recommended way to simulate user events when writing tests**.
```

Comment:
Mainly I think `to simulate user events` should be `to simulate user interactions` but I think it might be a minor improvement as:
```suggestion
For the above reasons, **the `user-event` library is recommended when writing tests for user interactions**.
```